src/ucs/memory/memtype_cache.c

```diff
@@ -127,19 +127,19 @@ static void ucs_memtype_cache_region_collect_callback(const ucs_pgtable_t *pgtab
 }
 
 UCS_PROFILE_FUNC_VOID(ucs_memtype_cache_update_internal,
-                      (address, size, mem_info, action),
+                      (memtype_cache, address, size, mem_info, action),
+                      ucs_memtype_cache_t *memtype_cache,
                       const void *address, size_t size,
                       const ucs_memory_info_t *mem_info,
                       ucs_memtype_cache_action_t action)
 {
-    ucs_memtype_cache_t *memtype_cache = ucs_memtype_cache_global_instance;
     ucs_memtype_cache_region_t *region, *tmp;
     UCS_LIST_HEAD(region_list);
     ucs_pgt_addr_t start, end, search_start, search_end;
     ucs_status_t status;
 
     if (memtype_cache == NULL) {
-        ucs_error("no memtype_cache instance found");
+        ucs_trace("no memtype_cache instance found");
```

Comment:
now this check does not make sense because memtype_cache is passed as a parameter to the function, better remove